File: crud-item-list/src/test/java/com/gurunars/crud_item_list/CheckerDeleteTest.java
package com.gurunars.crud_item_list;

import org.junit.Test;
import org.mockito.internal.util.collections.Sets;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class CheckerDeleteTest {

    private final ActionMoveDown<String> checkerDelete = new ActionMoveDown<>();
    private final List<String> all = Arrays.asList("one", "two");

    @Test
    public void selection_leadsToTrue() throws Exception {
        assertTrue(checkerDelete.canPerform(all, Sets.newSet("one")));
    }

    @Test
    public void noSelection_leadsToFalse() throws Exception {
        assertFalse(checkerDelete.canPerform(all, new HashSet<String>()));
    }

}

File: crud-item-list/src/test/kotlin/com/gurunars/crud_item_list/CheckerDeleteTest.kt
package com.gurunars.crud_item_list

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class CheckerDeleteTest {

    private val checkerDelete = ActionMoveDown<StringItem>()
    private val all = listOf("one", "two")

    @Test
    @Throws(Exception::class)
    fun selection_leadsToTrue() {
        checkerDelete.canPerform(all, setOf("one"), { assertTrue(it) })
    }

    @Test
    @Throws(Exception::class)
    fun noSelection_leadsToFalse() {
        checkerDelete.canPerform(all, setOf(), { assertFalse(it) })
    }
}
